### ISSUE TITLE
Error window on editing value in grid

### ISSUE THREAD

**Issue by fernandoval** [HeidiSQL/HeidiSQL#1107]
## Steps to reproduce this issue

1. Connect to a server/database;
2. Select a table into sidebar list;
3. Click on data tab;
4. Click on a column data;
5. Press enter to start edit mode;
6. Set a value;
7. Press enter to exit edit mode.

## Current behavior

The application error message is opened.

## Expected behavior

Stay in grid.

## Possible solution

Sorry!

## Environment

* HeidiSQL v 11.0.0.6061
* MySQL 5.7.30-0ubuntu0.18.04.1
* Ubuntu 18.04.4 LTS x86_64 / Wine 3.6
[bugreport.txt](https://github.com/HeidiSQL/HeidiSQL/files/4983200/bugreport.txt)

**Comment by ansgarbecker** [comment]
Wine 3.6 is outdated or? Please verify the issue persists with wine 4.0. Attention, newer v4 wine have problems with the tab key in Heidisql grids.

**Comment by fernandoval** [comment]
There is no Wine 4.x for Ubuntu 18.04.

The stable version of Wine is 5.0 and I can install it by PPA repo.

But the problem don't seem to be Wine.

This problem starts after the upgrade to night build 6057.

This error don't occours in v11.0.0.5919.

**Comment by ansgarbecker** [comment]
Also on Windows, when pressing Enter after editing, I get an access violation message in the bottom log panel (just a text message, not an application crash):

> /* #1634496361: Access violation at address 00000000005742AB in module 'heidisql.exe'. Read of address FFFFFFFFFFFFFFFF Message CharCode:13 Msg:256 */

I suppose you get a similar line there?

**Comment by fernandoval** [comment]
No. Don't have an error message in botton log panel. Only in dialog message that opens when pressing Enter after editing.

In error dialog the exception message says "Access violation at address 000000000041004E in module 'heidisql.exe'. Read of address 00000060FF000008."

But I realized that instead of pressing Enter, I click on another field after editing, the error does not occur.

**Comment by ansgarbecker** [comment]
Most probably related to #937 and probably also to my modification in bcce290a57a14797f82ca4f529fc37fb557487e7

**Comment by fernandoval** [comment]
There are almost two decades that I don't write a Pascal (Delphi) code.

But if I can help with anything, you can call me.

**Comment by zashme** [comment]
Same thing for wine 5.17

**Comment by fernandoval** [comment]
I did upgrade today to version 11.0.0.6099 and the issue stay alive.

**Comment by KR3KZ** [comment]
Same problem.
100% repro by trying to add a foreign key to a table.
Err: `/* #1634496361: Access violation at address 000000506A500000 in module 'heidisql.exe'. Execution of address 000000506A500000 Message CharCode:27 Msg:256 */`

**Comment by ansgarbecker** [comment]
@budiony I was sure this only happens through Wine, but you say the crash happens on Windows (10?) ? And it's not just the logged "_Access violation in xyz_" in the log panel?

**Comment by budiony** [comment]
> @budiony I was sure this only happens through Wine, but you say the crash happens on Windows (10?) ? And it's not just the logged "_Access violation in xyz_" in the log panel?

Yes indeed, Windows 10, I have attached log files and screenshots, but you deleted/closed my response as duplicate few weeks ago, I believe that this is problem of the edit field itself with combination of character encoding.

**Comment by ansgarbecker** [comment]
I suppose you mean #1179 - describing the same message appearing in the SQL log panel. Not a crash, just a debug message, but somehow the same cause. On Wine, the same message does not just appear in the log panel, but crashes the whole application.

**Comment by budiony** [comment]
Not sure about Wine, didn't use it myself under Linux. But again, it appears to me that the same reason could be the cause in both OS-es, given the fact that Wine comes with prebuild implementations of particular Windows DLLs, could be the same DLL(s) causing the problem, difference is that under emulation whole app crashes.

**Comment by fernandoval** [comment]
> I suppose you mean #1179 - describing the same message appearing in the SQL log panel. Not a crash, just a debug message, but somehow the same cause. On Wine, the same message does not just appear in the log panel, but crashes the whole application.

Since I upgraded to Wine 5.5 and HeidiSQL 11.0.0.6103, the crash windows did stop been showed.

Only a log message like this:

```
/* #1634496361: Access violation at address 0000001000000000 in module 'heidisql.exe'. Execution of address 0000001000000000 Message CharCode:13 Msg:256 */
```

**Comment by IKatsu** [comment]
While making a new table on Win10 connected to MariaDB 10.2.22
Heidisql 64bit 11.2.0.6213

/* #25: Access violation at address 00000000006CE623 in module 'heidisql.exe'. Read of address FFFFFFFFFFFFFFFF Message CharCode:0 Msg:514 */
/* #12: Abstract Error Message CharCode:0 Msg:514 */
/* #26: Abstract Error Message CharCode:0 Msg:514 */
/* #52: Access violation at address 000000000040EEE2 in module 'heidisql.exe'. Read of address FFFFFFFFFFFFFFFF Message CharCode:0 Msg:514 */
/* #1634496361: Access violation at address 0000005072F00000 in module 'heidisql.exe'. Execution of address 0000005072F00000 Message CharCode:13 Msg:256 */

(Timestamps on the log window would be nice too btw ;) )

**Comment by wlneary** [comment]
I can confirm that I am getting this on Win 10 connected to MariaDB 10.4.14, tested on HeidiSQL 11.3.6295 64 bit Portable, HeidiSQL 11.2.6213 64 bit Portable, and HeidiSQL 11.2.6213 64 bit.

To recreate, I can simply click on the field of an already existing row in the database and attempt editing it, or I can click on a row, press F2, and attempt to edit it. If I try to insert a new row, duplicate row, or duplicate without keys--into a table with a single primary key--I will get a "Grid Editing error" that looks like the attached image.
![error](https://user-images.githubusercontent.com/82120643/128566500-430bbf1a-be02-421f-9b8f-1350ecc9f20f.png)


```
/* #1634496361: Access violation at address 0000001000000000 in module 'heidisql.exe'. Execution of address 0000001000000000 Message CharCode:13 Msg:256 */
/* #1634496361: Access violation at address 0000001000000000 in module 'heidisql.exe'. Execution of address 0000001000000000 Message CharCode:13 Msg:256 */
```

**Comment by syphex97** [comment]
This issue was just presented to me a second ago, was working fine the past 3 days until now.

Version: 11.3.0.6295

**Comment by budiony** [comment]
LostCoins-Continue.bat stopped begin created in v.5.0.0, parameters used:
`C:\Users\bodi0\Desktop\LostCoins\LostCoins.exe --out C:\Users\bodi0\Desktop\LostCoins\Found.txt -t 0 -g -i 0,1,2 -x 256,256,256,256,256,256 -f C:\Users\bodi0\Desktop\LostCoins\puzzle160sorted.bin -r 1 -s <some starting address here> -z <some ending address here> -d 2 -n 5 -m 777` 

Note: <some starting address here> and <some ending address here> are just examples.

**Comment by PieterBreugelmans** [comment]
I can confirm that I'm also getting the same issue on Windows 10 using HeidiSQL 12.0.0.6468 (64-bit) against a MySQL database. Updating a value in the grid and press Enter key:

/* #1634496361: Access violation at address 000000000057769B in module 'heidisql.exe'. Read of address 0000000000000000 Message CharCode:13 Msg:256 */

**Comment by Carolus64** [comment]
Same issue on Windows 10 Pro Version 21H1 using HeidiSQL 12.1.0.6537 updating a grid value of a MySQL database.

/* #1634496361: Access violation at address 0000001000000000 in module 'heidisql.exe'. Execution of address 0000001000000000 Message CharCode:13 Msg:256 */

Anyway the database table is correctly modified, no other issue beside the display of message in the log pane to the bottom of application window.

**Comment by DHs2133** [comment]
I have a problem with HeidiSQL. I have this error: "Access violation at address 000000000040F142 in module 'heidisql.exe'. Read of address FFFFFFFFFFFFFFFF Message CharCode:0 Msg:514". 

I`m using Windows 10.

**Comment by RangerGH** [comment]
I am seeing this on 12.3.0.6589 Windows 10 Pro.

            
/* #1634496361: Access violation at address 00000000005E016B in module 'heidisql.exe'. Read of address 0000000000000000 Message CharCode:13 Msg:256 */

**Comment by olimexsmart** [comment]
I'm seeing this too, mainly while editing the grid during development. The problem is that the editing fails and it is a problem enough that I am considering other options. I am very sad since I've been an everyday user of HeidiSQL since 2017 basically :((( 

/* #1634496361: Access violation at address 00000056BC200000 in module 'heidisql.exe'. Execution of address 00000056BC200000 Message CharCode:13 Msg:256 */

**Comment by RangerGH** [comment]
Same here.  I really like HediSQL over other options I've tried (many of them) but if I can't edit it's not much use to me.

**Comment by olimexsmart** [comment]
I'm noticing that it is happening while interacting with any kind of grid really (like the one for editing columns). Other times it is harder to correlate to a particular action because maybe a lot of stuff is happening in the log. 

So I was reasoning about this, reaching my own, possibly useless, conclusions. While searching online I see that it is not something new (this very issue is close of being 3 years old) and at the same time not so common. The Google results are only 3 pages, mostly from the forum where this issue is often referenced by @ansgarbecker , so only 25 posts here in 2.5 years for such a breaking issue?

I recently changed machine and it's the first time that I notice this error. Could it be installation-related? I'll try to re-install and let you know.

**Comment by RangerGH** [comment]
I noticed the age on this issue, too which is why I posted - to add to it with more data.

I do not think it is installation related since I saw it on my previous machine.

I've since switched to DB Browser for SQLite but would like to use Heidi for other databases but with this issue I can't.

**Comment by ansgarbecker** [comment]
Please use the native Linux version shown on [the download page](https://www.heidisql.com/download.php). That should fix this issue.

Please note that the next major HeidiSQL release for Windows will drop support for Wine.
